src/i_flmusic.c

```diff
@@ -275,12 +278,31 @@ static void I_FL_ShutdownMusic(void)
 
 int I_FL_DeviceList(const char* devices[], int size)
 {
-    if (size > 0)
+    int i;
```

Comment:
We need to add the SNDFONT lump if it's available.

Reply:
Or somehow suppress this whole procedure if the WAD author has already chosen a soundfont for us. 😉

Reply:
Currently the SNDFONT lump takes precedence and the user's choice is ignored.

Reply:
But the soundfont choice, albeit ignored, will still appear in the menu?

Reply:
> But the soundfont choice, albeit ignored, will still appear in the menu?

Yes.

Reply:
Then `I_FL_DeviceList()` should rather set ` devices[0] = "FluidSynth (sndfont lump)"` and  `return 1` in that case.

Reply:
> Then `I_FL_DeviceList()` should rather set ` devices[0] = "FluidSynth (sndfont lump)"` and `return 1` in that case.

But then it will alter `midi_player` config variable.

I suggest disabling the "Midi Player" menu item and forcing the Fluidsynth if SNDFONT lump is found.

Reply:
Yes, okay. But if we add a new soundfont to the directory between two sessions, this will also change the meaning of the `midi_player` config variable, right?

Reply:
Correct, but I think the user will expect this. If some PWAD changes midi player settings it will be annoying.

Reply:
Yes, right.